### ISSUE TITLE
Stream can't be closed

### ISSUE THREAD

**Issue by TristisOris** [slotix/dbconvert-streams-public#13]
v0.8.5
After finished migration:
```
curl --request DELETE --url http://127.0.0.1:8020/api/v1/streams
curl --request GET --url http://127.0.0.1:8020/api/v1/streams/stat
{"streamID":"2ZCk6GC8AwWwLJcAGp2PHHlaqje","source":{"Success events":1170616,"Failed events":0,"Data Size":0,"Elapsed":"31m37.111648512s","Started":"2023-12-07T10:14:32+03:00","Status":"FINISHED"},"target":{"Success events":1170616,"Failed events":0,"Data Size":0,"Elapsed":"31m37.143094256s","Started":"2023-12-07T10:14:32+03:00","Status":"FINISHED"}}
```

During migration
```
curl --request GET --url http://127.0.0.1:8020/api/v1/streams/stat
{"streamID":"2ZCyXEd7pmpJi9Bo3xKe46ATNK9","source":{"Success events":220667955,"Failed events":0,"Data Size":0,"Elapsed":"41m49.794901781s","Started":"2023-12-07T12:13:15+03:00","Status":"FINISHED"},"target":{"Success events":50501569,"Failed events":0,"Data Size":0,"Elapsed":"0s","Started":"2023-12-07T12:13:15+03:00","Status":"RUNNING"}}

curl -curl --request DELETE --url http://127.0.0.1:8020/api/v1/streams
curl --request GET --url http://127.0.0.1:8020/api/v1/streams/stat
{"streamID":"2ZCyXEd7pmpJi9Bo3xKe46ATNK9","source":{"Success events":220667955,"Failed events":0,"Data Size":0,"Elapsed":"41m49.794901781s","Started":"2023-12-07T12:13:15+03:00","Status":"FINISHED"},"target":{"Success events":50501569,"Failed events":0,"Data Size":0,"Elapsed":"0s","Started":"2023-12-07T12:13:15+03:00","Status":"RUNNING"}}

[2023/12/07 13:40:30] [info] target/target.go:114 [target] Status: RUNNING, Success events: 50501569 Data size: 12.2 GB
[2023/12/07 13:40:37] [error] target/consumers.go:106 nats: consumer deleted
[2023/12/07 13:40:37] [error] target/consumers.go:106 nats: consumer deleted
[2023/12/07 13:40:37] [error] target/consumers.go:52 nats: consumer deleted
[2023/12/07 13:40:37] [error] target/consumers.go:52 nats: consumer deleted
[2023/12/07 13:40:37] [error] target/consumers.go:52 nats: consumer deleted
[2023/12/07 13:40:37] [error] target/consumers.go:52 nats: consumer deleted
[2023/12/07 13:40:37] [error] target/consumers.go:52 nats: consumer deleted
[2023/12/07 13:40:37] [error] target/consumers.go:52 nats: consumer deleted
[2023/12/07 13:40:37] [error] target/consumers.go:52 nats: consumer deleted
[2023/12/07 13:40:37] [error] target/consumers.go:52 nats: consumer deleted
[2023/12/07 13:40:37] [error] target/consumers.go:52 nats: consumer deleted
[2023/12/07 13:40:37] [error] target/consumers.go:52 nats: consumer deleted
[2023/12/07 13:40:40] [info] target/target.go:114 [target] Status: RUNNING, Success events: 50501569 Data size: 12.2 GB
[2023/12/07 13:40:50] [info] target/target.go:114 [target] Status: RUNNING, Success events: 50501569 Data size: 12.2 GB
```

Need to stop all workers to run new task.

**Comment by slotix** [comment]
@TristisOris  [Version 0.9.0](https://stream.dbconvert.com/releases-2024#version-0-9-0) addresses and resolves the reported issue.

**Comment by TristisOris** [comment]
Looks it not fully fixed.

During active migration `curl --request DELETE --url http://127.0.0.1:8020/api/v1/streams`
Then
```
curl --request POST --url http://127.0.0.1:8020/api/v1/streams\?file=test-v2.json
{"error":"failed to initialize a new stream while another stream is active (in INIT or RUNNING state)."}
```

```
 ./dbs-api
DBConvert Streams API : starting at 0.0.0.0:8020
Version: v0.9.1 Build time: 2024-01-12_11:26:08
Find more information at https://stream.dbconvert.com
open connections: no such file or directory
open streams: no such file or directory
[2024/02/01 12:06:07] [info] eventhub/eventhub.go:67 connected to EventHub:  {"address": "nats://127.0.0.1:4222"}
[2024/02/01 12:06:16] [info] dbs-api/stream.go:407 created stream: "2bl8bVosooP17puxtiJN7XELKfP"
[2024/02/01 12:09:13] [error] dbs-api/stream.go:73 failed to initialize a new stream while another stream is active (in INIT or RUNNING state).
```

Reader is still running
```
[2024/02/01 12:06:21] [info] source/source.go:317 [source] number of consumers available for stream 2bl8bVosooP17puxtiJN7XELKfP: 3
[2024/02/01 12:06:31] [info] [source] RUNNING, 640 events, 89.2 MB, Rate: 8.9 MB/s
[2024/02/01 12:06:41] [info] [source] RUNNING, 1,480 events, 161.3 MB, Rate: 7.2 MB/s
[2024/02/01 12:06:51] [info] [source] RUNNING, 1,960 events, 244.1 MB, Rate: 8.3 MB/s
[2024/02/01 12:07:01] [info] [source] RUNNING, 2,880 events, 325.6 MB, Rate: 8.2 MB/s
[2024/02/01 12:07:11] [info] [source] RUNNING, 3,440 events, 412.1 MB, Rate: 8.6 MB/s
[2024/02/01 12:07:21] [info] [source] RUNNING, 4,080 events, 462.0 MB, Rate: 5.0 MB/s
[2024/02/01 12:07:31] [info] [source] RUNNING, 4,640 events, 549.4 MB, Rate: 8.7 MB/s
[2024/02/01 12:07:41] [info] [source] RUNNING, 5,320 events, 602.3 MB, Rate: 5.3 MB/s
[2024/02/01 12:07:51] [info] [source] RUNNING, 5,680 events, 668.4 MB, Rate: 6.6 MB/s
[2024/02/01 12:08:01] [info] [source] RUNNING, 6,560 events, 743.8 MB, Rate: 7.5 MB/s
[2024/02/01 12:08:11] [info] [source] RUNNING, 7,360 events, 838.5 MB, Rate: 9.5 MB/s
```

Writers closed
```
[2024/02/01 12:08:10] [error] target/target.go:361 parsing time "2020-01-27" as "2006-01-02T15:04:05Z07:00": cannot parse "" as "T"
[2024/02/01 12:08:10] [error] target/target.go:361 parsing time "2020-09-14" as "2006-01-02T15:04:05Z07:00": cannot parse "" as "T"
[2024/02/01 12:08:10] [info] [target] RUNNING, Rate: 9.8 MB/s, 7,320 failed
[2024/02/01 12:08:12] [error] target/target.go:361 parsing time "2020-03-18" as "2006-01-02T15:04:05Z07:00": cannot parse "" as "T"
[2024/02/01 12:08:13] [info] [target] STOPPED, Rate: 1.4 MB/s, 7,520 failed, Elapsed: 1m51.991646795s
```

**Comment by slotix** [comment]
Sometimes, in the case where the target has a 'Status_FAILED,' it fails to correctly finish its workers and apply the 'failed' status to this target node. Consequently, a new stream can't be initiated from the API due to this ghosted target instance. This issue needs to be addressed and fixed accordingly.

**Comment by slotix** [comment]
Fixed in the [version 0.10.3](https://stream.dbconvert.com/releases-2024#version-0-10-3)

**Comment by slotix** [comment]
Sometimes, in the case of failure during the creation of tables and indexes on the target database, DBS reader and writer services fail to update their statuses correctly. As a result, the API is unable to accept new stream configurations.

Services must notify other service instances about their 'failing' status accordingly.

**Comment by slotix** [comment]
Fixed in [version 0.10.9](https://stream.dbconvert.com/releases-2024#version-0-10-9)
